internal/api/schemas/v1.yaml

```diff
@@ -611,6 +779,55 @@ components:
           type: string
         value:
           type: string
+    TCPRoutePage:
+      type: object
+      required:
+      - routes
+      properties:
+        routes:
+          type: array
+          items:
+            "$ref": "#/components/schemas/TCPRoute"
+        meta:
+          type: object
+    TCPRoute:
+      type: object
+      required:
+      - name
+      - namespace
+      - gateways
```

Comment:
Analogous to `ParentRefs`, K8s doesn't require these be set. Thoughts?

https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.CommonRouteSpec

Reply:
Same as the HTTPRoute stuff -- I'd prefer to keep this as required at least initially. We can always make the API more permissive if we so desire without it being a breaking change, but can't really go the other direction without a break -- but not entirely sure why the refs aren't required for K8s and I think(?) it has something to do with plans for later route delegation.